### PR TITLE
Add `Getting Started` page 

### PR DESCRIPTION
**Pull request by b-reyes**
This PR addresses #325 by creating a flowchart for both HPC and Cloud resources documentation. In addition to this major change, there was a large amount of reorganization and cleaning up of documentation. A summary of all changes is provided below. 

## Refactor and Reorganize Documentation
- **File Naming and Structure**
  - Rename and move directories and files for clarity (e.g., `access` to `getting_started`, `Petalibrary` to `storage/petalibrary` and name it `mounting_images`).

- **Tab Format Updates**
  - Update tabs in various markdown files to a new format for consistency across `logging-in.md`, `OpenMP-C.md`, `OpenMP-Fortran.md`, `MPI-C.md`, `MPI-Fortran.md`, `MPIBestpractices.md`, `coding-best-practices.md`, `spack.md`, `petalibrary.md`, `batch-jobs.md`,`interactive-jobs.md`, `alpine-hardware`, `slurm_directive_ex`, and `filesystems` pages.
  - Remove all old css and libraries associated with old tabs. Ensure that spinx-design is a requirement now. 

- **Document Clean-up**
  - Remove unused directories: `XSEDE`, `Templates`, `Loading-Modules`, `Job-Submissions`, `Interactive-Jobs`, and `Logging-in`.
  - Clean up `rmacc` folder and remove obsolete images.

## Automation and Scripting Enhancements
- **Scripts and Tools**
  - Create a Python script for generating SVG flowcharts from dot files.
  - Develop a script to check tab references and integrate it into the Makefile and RTD yaml.
  - Develop a script to check dot file references and integrate it into the Makefile and RTD yaml.

- **Graph and Flowchart Improvements**
  - Adjust color schemes and dimensions for flowcharts, ensuring SVGs are responsive.
  - Enhance flowchart generation with a Python script to clean generated SVGs.
  - Add all necessary requirements to `conf.py` and RTD yaml

## Content Adjustments
- **Content Organization**
  - Reorganize the FAQ page for clarity (partially addresses #377)
  - Add all Duo content to FAQ page

- **Enhancements and Tips**
  - Add introductory paragraphs and tips across various pages to improve user experience.
  - Update references to ensure they point to the correct sections.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,116 +1,30 @@
-# Research Computing User Guide
+# CU Research Computing User Guide
 
-Documentation covering the use of Research Computing resources.
+Welcome to CU Research Computing's (CURC's) user guide! CURC offers a variety of services encompassing High Performance Computing (HPC), cloud computing assistance, and storage solutions. Many of these services are available to users associated with CU Boulder, Anschutz Medical Campus (AMC), Colorado State University (CSU), and institutions who are members of the Rocky Mountain Advanced Computing Consortium (RMACC).
```
Comment (SchneiderCode) [inline docs/index.md:3]: Grammar Change "institutions who are members" to "institutions that are members"

File: docs/getting_started/navigating_docs.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Navigating CURC Documentation
+
+CU Research Computing (CURC) offers a variety of services encompassing High Performance Computing (HPC), cloud computing assistance, and storage solutions. Many of these services are available to users associated with CU Boulder, Anschutz Medical Campus (AMC), Colorado State University (CSU), and institutions who are members of the Rocky Mountain Advanced Computing Consortium (RMACC). Our team understands that navigating the various services we provide can be overwhelming for both new and experienced users. To aid users, we have constructed flowcharts that outline the core areas of our documentation that every user on our system should become familiar with. Please refer to the section [High Performance Computing Flowchart](#high-performance-computing-flowchart) for a diagram on how to navigate our HPC systems (clusters) and storage solutions. For a diagram on the various cloud computing resources we can assist you with, please refer to the [Cloud Computing Flowchart](#cloud-computing-flowchart) section.
```
Comment (SchneiderCode) [inline docs/getting_started/navigating_docs.md:3]: Grammar Change "institutions who are members" to "institutions that are members"

Comment (SchneiderCode) [inline docs/getting_started/navigating_docs.md:3]:
```suggestion
CU Research Computing (CURC) offers a variety of services encompassing High Performance Computing (HPC), cloud computing assistance, and storage solutions. Many of these services are available to users associated with CU Boulder, Anschutz Medical Campus (AMC), Colorado State University (CSU), and institutions that are members of the Rocky Mountain Advanced Computing Consortium (RMACC). Our team understands that navigating the various services we provide can be overwhelming for both new and experienced users. To aid users, we have constructed flowcharts that outline the core areas of our documentation that every user on our system should become familiar with. Please refer to the section [High Performance Computing Flowchart](#high-performance-computing-flowchart) for a diagram on how to navigate our HPC systems (clusters) and storage solutions. For a diagram on the various cloud computing resources we can assist you with, please refer to the [Cloud Computing Flowchart](#cloud-computing-flowchart) section. 
```

File: docs/index.md (hunk 1 above)
Comment (SchneiderCode) [inline docs/index.md:3]:
```suggestion
Welcome to CU Research Computing's (CURC's) user guide! CURC offers a variety of services encompassing High Performance Computing (HPC), cloud computing assistance, and storage solutions. Many of these services are available to users associated with CU Boulder, Anschutz Medical Campus (AMC), Colorado State University (CSU), and institutions that are members of the Rocky Mountain Advanced Computing Consortium (RMACC). 
```

File: docs/getting_started/navigating_docs.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## High Performance Computing Flowchart
+
+At CURC we host two HPC clusters, Alpine and Blanca, in addition to storage solutions. To utilize these resources there are several items that are important to be aware of. Although our documentation covers a variety of other topics, below we provide a flowchart that we believe identifies the key areas users of these services should read.
```
Comment (SchneiderCode) [inline docs/getting_started/navigating_docs.md:7]:
```suggestion
At CURC, we host two HPC clusters, Alpine and Blanca, along with various storage solutions. To make the most of these resources, there are several crucial points to consider. While our documentation addresses many topics, we've included a flowchart below that highlights the key areas users should focus on.
```

File: docs/getting_started/navigating_docs.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## High Performance Computing Flowchart
+
+At CURC we host two HPC clusters, Alpine and Blanca, in addition to storage solutions. To utilize these resources there are several items that are important to be aware of. Although our documentation covers a variety of other topics, below we provide a flowchart that we believe identifies the key areas users of these services should read. 
+
+```{tip}
+In the flowchart presented below, you can easily navigate to the corresponding documentation page by clicking on the associated graph node. All nodes that change color when hovered over are clickable.
```
Comment (SchneiderCode) [inline docs/getting_started/navigating_docs.md:10]:
```suggestion
In the flowchart below, you can easily navigate to the corresponding documentation page by clicking on the relevant node. Nodes that change color when hovered over are clickable.
```

File: docs/getting_started/navigating_docs.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+.. raw:: html
+    :file: ../../graphviz_flowcharts/generated_images/getting_started_hpc_flowchart.svg
+```
+
+## Cloud Computing Flowchart
+
+The Cloud Foundations Service (CFS) at CURC provides an additional layer of support between the cloud provider and the customer. The purpose of CFS is to manage the integration of the cloud with CU standards. Specifically, CFS provides assistance with the provisioning of public cloud accounts (AWS, Azure, GCP), cost estimation, billing against CU funds, CU internal network connection, consulting, and high-level architectural guidance. It should be noted that CFS customers are expected to manage their applications, data, and use of cloud services. Below we provide a flowchart that we believe identifies the key areas users of these services should read. 
+
+```{tip}
+In the flowchart presented below, you can easily navigate to the corresponding documentation page by clicking on the associated graph node. All nodes that change color when hovered over are clickable.
```
Comment (SchneiderCode) [inline docs/getting_started/navigating_docs.md:23]:
```suggestion
In the flowchart below, you can easily navigate to the corresponding documentation page by clicking on the relevant node. Nodes that change color when hovered over are clickable.
```

File: docs/getting_started/logging-in.md [hunk 6]
```diff
@@ -0,0 +1,250 @@
+# Logging In
+
+```{tip}
+Do you need help navigating our documentation? For new and experienced users, we recommend reviewing our [Navigating CURC Documentation](./navigating_docs.md) page, which includes flowcharts for core areas of our documentation that all users should be aware of.
```
Comment (SchneiderCode) [inline docs/getting_started/logging-in.md:4]:
```suggestion
Need help navigating our documentation? Whether you're new or experienced, we recommend reviewing our [Navigating CURC Documentation](./navigating_docs.md) page. It includes flowcharts covering the core areas all users should be aware of.
```

File: docs/getting_started/logging-in.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
+# Logging In
+
+```{tip}
+Do you need help navigating our documentation? For new and experienced users, we recommend reviewing our [Navigating CURC Documentation](./navigating_docs.md) page, which includes flowcharts for core areas of our documentation that all users should be aware of. 
+```
+
+## Getting a CURC account
+
+All individuals who would like to access CU Research Computing (CURC) HPC systems (Alpine and Blanca), Open OnDemand, and storage solutions must have an associated CURC account. CURC systems are utilized by a variety of institutions including CU Boulder, Anschutz Medical Campus (AMC), Colorado State University (CSU), and institutions who are members of the Rocky Mountain Advanced Computing Consortium (RMACC). Given CURC systems host a variety of institutions, each institution has its own method for creating a CURC account. To simplify this process, below we provide tabs that specify instructions on getting a CURC account based on the institution you are affiliated with.
```
Comment (SchneiderCode) [inline docs/getting_started/logging-in.md:9]:
```suggestion
All individuals who would like to access CU Research Computing (CURC) HPC systems (Alpine and Blanca), Open OnDemand, and storage solutions must have an associated CURC account. CURC systems are utilized by a variety of institutions including CU Boulder, Anschutz Medical Campus (AMC), Colorado State University (CSU), and institutions that are members of the Rocky Mountain Advanced Computing Consortium (RMACC). Given CURC systems host a variety of institutions, each institution has its own method for creating a CURC account. To simplify this process, below we provide tabs that specify instructions on getting a CURC account based on the institution you are affiliated with. 
```

File: docs/getting_started/logging-in.md [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+- Once your account is provisioned, please wait 15 minutes before signing in
+- If you are having issues with DUO, please consult our [Duo Frequently Asked Questions](./faq.md#duo-multi-factor-authentication) section
+````
+
+`````
+
+`````{tab-item} CSU
+:sync: logging-in-csu
+
+The process for obtaining a CURC account for CSU users is straightforward. Simply navigate to our [Account Request](https://rcamp.rc.colorado.edu/accounts/account-request/create/organization) page, select **"Colorado State University"**, fill out the provided information, and select **"Verify & Continue"**. Once the form has been completed and submitted, an automatic process will provision your account. Once you've received a CURC account, you will need to enroll in Duo two-factor authentication service. For more information on setting up Duo, please visit CSU's [Duo Two Factor Authentication](https://it.colostate.edu/duo-two-factor-authentication/) page. For additional information on using CURC resources as a CSU user, please consult [CSU's Getting Started with Alpine](https://it.colostate.edu/research-computing-and-cyberinfrastructure/compute/get-started-with-alpine/) documentation.
```
[the remark this answers is not in the document]
Comment (SchneiderCode) [reply docs/getting_started/logging-in.md:31]: Updated the link - the old one went to a 404 Page not found.

File: docs/getting_started/logging-in.md [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+Similar to obtaining an account, the process of logging in to CURC resources can vary based on the institution you are affiliated with. In the tab sections below we provide details on how each institution can gain access to our login nodes.
+
+(tabset-ref-get-access)=
+`````{tab-set}
+:sync-group: tabset-logging-in
+
+````{tab-item} CU Boulder
+:sync: logging-in-ucb
+
+For CU Boulder users, the following methods are available for logging in to CURC resources:
```
Comment (SchneiderCode) [inline docs/getting_started/logging-in.md:107]:
```suggestion
For CU Boulder users, the following methods are available for logging into CURC resources:
```

File: docs/getting_started/logging-in.md [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+```{important}
+- Once your account is provisioned, please wait 15 minutes before signing in. 
+```
+
+`````
+
+`````{tab-item} AMC
+:sync: logging-in-amc
+
+Currently, AMC users must go through various steps to create a CURC account. These steps are outlined below:
```
Comment (SchneiderCode) [inline docs/getting_started/logging-in.md:42]:
```suggestion
Currently, AMC users must go through the following steps to create a CURC account: 
```

File: docs/getting_started/logging-in.md [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+    - See our [Open OnDemand (Browser Based HPC Portal)](../open_ondemand/index.md) documentation for more information
+````
+
+````{tab-item} CSU
+:sync: logging-in-csu
+
+For CSU users, the following methods are available for logging in to CURC resources:
+- Utilizing Secure Shell (SSH) from your computer's terminal to `ssh` into a login node
+    - For more information, on how to do this, see [Terminal access for different operating systems](#terminal-access-for-different-operating-systems) below
+    - For CSU specific guides for this process, please refer to the **"Remote Login"** section of CSU's [Get Started with Alpine](https://it.colostate.edu/research-computing-and-cyberinfrastructure/compute/get-started-with-alpine/#) page
```
Comment (SchneiderCode) [inline docs/getting_started/logging-in.md:120]:
```suggestion
    - For CSU specific guides for this process, please refer to the **"Remote Login"** section of CSU's [Get Started with Alpine](https://www.research.colostate.edu/rit/research-computing-cyberinfrastructure/get-started-with-alpine/) page
```

**Review by SchneiderCode — APPROVED** [review]
I added a few word changes, but the only major change was resolving a broken external link. At some point the CSU "Getting Started with Alpine" link changed - I added in the new link.

File: docs/getting_started/navigating_docs.md (hunk 3 above)
Comment (b-reyes) [reply docs/getting_started/navigating_docs.md]: I modified this sentence slightly from what you provide, but kept a lot of the elements. Thank you for the suggestions!

**Pull request merged**